internal/cri/server/restart.go

```diff
@@ -195,6 +195,25 @@ func (c *criService) recover(ctx context.Context) error {
 		return err
 	}
 
+	for _, container := range c.containerStore.List() {
```

Comment:
Let's merge it into L175 function.

And could you please add e2e test for this as well? Thanks.

Reply:
L175 c.loadContainer can we use another pr https://github.com/containerd/containerd/pull/10389/files?
this pr  I refer L 153, for e2e do you have some exapmples ?

Reply:
L153 will add some extra funcs will make containerd start slower.

Reply:
ping @fuweid

Reply:
Hi @ningmingxiao sorry for late reply.

What I think is that the `loadContainer` function can be like

```go
func (c *criService) loadContainer(ctx context.Context, cntr containerd.Container) (containerstore.Container, <-chan ExitStatus, error) {
...
var exitCh <-chan ExitStatus
...
exitCh, err = t.Wait(ctrdutil.NamespacedContext())
if err != nil {
    ...
   exitCh = nil
}
...
}
```

So for the caller, it should be like

```go
cntr, exitCh, err := c.loadContainer(ctx2, container)
...

c.containerStore.Add(cntr)

c.startSandboxExitMonitor(context.Background(), cntr.ID, exitCh)
```

What do you think?